File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/chainofresponsibility/cache/CacheHandler.java
package com.bigboxcode.designpattern.chainofresponsibility.cache;

public abstract class CacheHandler {
    public CacheHandler nextCacheHandler;

    public CacheHandler(CacheHandler nextCacheHandler) {
        this.nextCacheHandler = nextCacheHandler;
    }

    public abstract void handleRequest(Data data);
}

File: design-patterns/go/chain_of_responsibility/cache/cache_handler.go
// cache_handler.go

package main

type CacheHandler interface {
	HandleRequest(data Data)
}

// type CacheHandler struct {
// 	nextCacheHandler *CacheHandler
// }

// func NewCacheHandler(nextCacheHandler *CacheHandler) (cacheHandler *CacheHandler) {
// 	cacheHandler = &CacheHandler{}
// 	cacheHandler.nextCacheHandler = nextCacheHandler
// 	return
// }

// func (cacheHandler *CacheHandler) HandleRequest(data *Data) {
// }